subsys/bluetooth/mesh/settings.c

```diff
@@ -98,12 +104,10 @@ void bt_mesh_settings_store_schedule(enum bt_mesh_settings_flag flag)
 
 	if (atomic_get(pending_flags) & NO_WAIT_PENDING_BITS) {
 		timeout_ms = 0;
-	} else if (atomic_test_bit(pending_flags,
-				   BT_MESH_SETTINGS_RPL_PENDING) &&
-		   (!(atomic_get(bt_mesh.flags) & GENERIC_PENDING_BITS) ||
-		    (CONFIG_BT_MESH_RPL_STORE_TIMEOUT <
-		     CONFIG_BT_MESH_STORE_TIMEOUT))) {
-		timeout_ms = CONFIG_BT_MESH_RPL_STORE_TIMEOUT * MSEC_PER_SEC;
+	} else if (atomic_test_bit(pending_flags, BT_MESH_SETTINGS_RPL_PENDING) &&
+		   !(atomic_get(bt_mesh.flags) & GENERIC_PENDING_BITS) &&
+		   !(RPL_STORE_TIMEOUT < CONFIG_BT_MESH_STORE_TIMEOUT)) {
```

Comment:
I think this || should still be here. If RPL is pending and the RPL timeout is shorter than the general timeout, it should use this, right? Perhaps a simpler version would be 
```c
else if (atomic_test_bit(pending_flags, BT_MESH_SETTINGS_RPL_PENDING)) {
        timeout_ms = MIN(RPL_STORE_TIMEOUT, CONFIG_BT_MESH_STORE_TIMEOUT) * MSEC_PER_SEC;
} else {
        timeout_ms = CONFIG_BT_MESH_STORE_TIMEOUT * MSEC_PER_SEC;
}
```
might need an additional `RPL_STORE_TIMEOUT != -1` here if you make the change I suggested in kconfig, but it'd still seem cleaner than the original, I think

Reply:
Oh, right, I messed this up. Will fix.